lib/Parser.js

```diff
@@ -659,6 +659,7 @@ function parseExpr(o, literals, result, start, useBrackets) {
     isTop = true;
   }
   for (var i = start, len = o.str.length; i < len; ++i) {
+    var isMessageId = isEnvelope && result.length == 9;
```

Comment:
From the IMAP Spec: https://tools.ietf.org/html/rfc3501

The fields of the envelope structure are in the following order: date, subject, from, sender, reply-to, to, cc, bcc, in-reply-to, and message-id.

If processing innerResult of ENVELOPE, and 9 items already exist, then code is currently working on the message-id param.